### PR TITLE
External variable cli

### PR DESCRIPTION
**Pull request by ghislainfourny**

### CODE REVIEW

**Comment by ghislainfourny** [comment]
@CanBerker this PR introduces basic support for passing external variables from the CLI. It would be great if you could apply IntelliJ to the changed parser files. Thanks a lot!

File: src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java [hunk 1]
```diff
@@ -57,10 +57,10 @@
 
 public class JsoniqQueryExecutor {
     public static final String TEMP_QUERY_FILE_NAME = "Temp_Query";
-    private SparksoniqRuntimeConfiguration configuration;
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java]: I'm glad to see this rename

**Comment by CanBerker** [comment]
#597 is created to format generated files. Feel free to merge it if it looks good.

**Comment by ghislainfourny** [comment]
@CanBerker has the generated files with wildcard removal and spotless applied. The diff is big, but if we agree to stick to this convention, diffs may be small again in the future.

File: src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import org.rumbledb.expressions.AbstractNodeVisitor;
+import org.rumbledb.expressions.Node;
+import org.rumbledb.expressions.module.VariableDeclaration;
+import org.rumbledb.items.ItemFactory;
+import org.rumbledb.types.SequenceType.Arity;
+
+import sparksoniq.semantics.DynamicContext;
+
+/**
+ * Static context visitor implements a multi-pass algorithm that enables function hoisting
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java]: This comment should be removed. Feel free to add a documentation for the DynamicContextVisitor's purpose as well.

File: src/main/java/org/rumbledb/compiler/StaticContextVisitor.java [hunk 3]
```diff
@@ -288,4 +289,18 @@ public StaticContext visitTypeSwitchExpression(TypeSwitchExpression expression,
     }
     // endregion
 
+    // region FLWOR vars
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/compiler/StaticContextVisitor.java]: I think this region specifier may have been accidentally copied.

File: src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        List<Item> values = new ArrayList<>();
+        if (value != null) {
+            Item item = ItemFactory.getInstance().createStringItem(value);
+            values.add(item);
+            if (
+                variableDeclaration.getSequenceType().isEmptySequence()
+                    || !item.isTypeOf(variableDeclaration.getSequenceType().getItemType())
+            ) {
+                throw new UnexpectedTypeException(
+                        "External variable value is empty and does not match the expected type.",
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java]: It seems to me that the phrasing of the error message (empty 'and' type mistmatch) does not match the if condition (empty 'or' type mismatch) . Is this intentional?

File: src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java (hunk 2 above)
Comment (ghislainfourny) [reply src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java]: Done

File: src/main/java/org/rumbledb/compiler/StaticContextVisitor.java (hunk 3 above)
Comment (ghislainfourny) [reply src/main/java/org/rumbledb/compiler/StaticContextVisitor.java]: Done

File: src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java (hunk 4 above)
Comment (ghislainfourny) [reply src/main/java/org/rumbledb/compiler/DynamicContextVisitor.java]: Done

File: src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java (hunk 1 above)
Comment (ghislainfourny) [reply src/main/java/org/rumbledb/cli/JsoniqQueryExecutor.java]: I also renamed the context package. :)

**Pull request merged**